### PR TITLE
Move pass pattern to ov

### PR DESCRIPTION
**Pull request by ilyachur**
PR depends on https://github.com/openvinotoolkit/openvino/pull/7240

### CODE REVIEW

File: ngraph/core/include/openvino/pass/graph_rewrite.hpp [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        auto pass_config = get_pass_config();
+
+        for (auto& matcher : pass->m_matchers) {
+            pass->set_pass_config(pass_config);
+            m_matchers.push_back(matcher);
+        }
+    }
+
+    OPENVINO_DEPRECATED("Use MatcherPass instead")
+    void add_matcher(const std::shared_ptr<pattern::Matcher>& m,
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/pass/graph_rewrite.hpp:199]: can we remove it finally?

File: ngraph/core/include/ngraph/pattern/op/true.hpp [hunk 2]
```diff
@@ -6,21 +6,12 @@
 
 #include "ngraph/node.hpp"
 #include "ngraph/pattern/op/pattern.hpp"
+#include "openvino/pass/pattern/op/true.hpp"
```
Comment (ilya-lavrenov) [inline ngraph/core/include/ngraph/pattern/op/true.hpp:9]: do we need `ngraph/xxx` headers then?

**Review by ilya-lavrenov — APPROVED** [review]

File: ngraph/core/include/ngraph/pattern/op/true.hpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/pattern/op/true.hpp:9]: Yes, I don't remove these headers in order to save backward compatibility for cases when:
user includes `ngraph/xxx.hpp` header which includes `ngraph/yyy.hpp` header and user uses functionality from both headers.
If I remove these headers, it means that I can break something in user application.

File: ngraph/core/include/openvino/pass/graph_rewrite.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/include/openvino/pass/graph_rewrite.hpp:199]: It can be backward incompatible for our old users.

Comment (ilya-lavrenov) [reply ngraph/core/include/openvino/pass/graph_rewrite.hpp:199]: why? it's deprecated for several releases

Comment (ilyachur) [reply ngraph/core/include/openvino/pass/graph_rewrite.hpp:199]: Hmm, yes, it is a valid comment. So I propose to remove these deprecated methods in the separate PR.

Comment (ilya-lavrenov) [reply ngraph/core/include/openvino/pass/graph_rewrite.hpp:199]: ok, let's merge then

File: ngraph/core/include/ngraph/pass/graph_rewrite.hpp [hunk 3]
```diff
@@ -10,240 +10,17 @@
...
-
-private:
-    size_t m_num_iters;
-
-    std::vector<std::shared_ptr<ngraph::pass::MatcherPass>> m_matchers;
-};
+using ov::pass::BackwardGraphRewrite;
+using ov::pass::GraphRewrite;
+using ov::pass::MatcherPass;
+using ov::pass::RecurrentGraphRewrite;
```
Comment (GlebKazantaev) [inline ngraph/core/include/ngraph/pass/graph_rewrite.hpp:24]: Let's keep it only in legacy API.

File: ngraph/core/include/ngraph/pass/pass.hpp [hunk 4]
```diff
@@ -13,105 +13,32 @@
 #include "ngraph/node.hpp"
 #include "ngraph/pass/pass_config.hpp"
 #include "ngraph/util.hpp"
+#include "openvino/pass/pass.hpp"
 
-namespace ngraph {
+namespace ov {
 namespace pass {
-enum class PassProperty : uint32_t {
```
Comment (GlebKazantaev) [inline ngraph/core/include/ngraph/pass/pass.hpp:19]: Let's keep pass property in legacy API

File: ngraph/core/include/ngraph/pass/pass.hpp [hunk 5]
```diff
@@ -13,105 +13,32 @@
...
+}  // namespace ov
+namespace ngraph {
+namespace pass {
+using ov::pass::FunctionPass;
+using ov::pass::FusionType;
+using ov::pass::FusionTypeMask;
+using ov::pass::Manager;
+using ov::pass::PassBase;
+using ov::pass::PassProperty;
+using ov::pass::PassPropertyMask;
```
Comment (GlebKazantaev) [inline ngraph/core/include/ngraph/pass/pass.hpp:33]: keep in legacy

File: ngraph/core/include/ngraph/pattern/matcher.hpp [hunk 6]
```diff
@@ -16,255 +16,21 @@
...
-    Output<Node> m_initial_pattern;
-    Output<Node> m_pattern;
-    std::shared_ptr<Node> m_recurrent_pattern;
-    const std::set<std::shared_ptr<Node>> m_correlated_patterns;
-    RPatternValueMap m_matches;
-    Output<Node> m_match_root;
-};
+using ov::pass::pattern::Matcher;
+using ov::pass::pattern::MatcherState;
+using ov::pass::pattern::RecurrentMatcher;
```
Comment (GlebKazantaev) [inline ngraph/core/include/ngraph/pattern/matcher.hpp:34]: Let's keep it in legacy API

File: ngraph/core/include/openvino/pass/graph_rewrite.hpp (hunk 1 above)
Comment (GlebKazantaev) [reply ngraph/core/include/openvino/pass/graph_rewrite.hpp:199]: It is still used by some OV components actually

File: ngraph/core/include/openvino/pass/pass.hpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    //`DIFFERENTIABLE_FUSIONS` produce ops that support autodiff
+    // i.e. implement `generate_adjoints`
+    DIFFERENTIABLE_FUSIONS = 0x1,
+    REGULAR_FUSIONS = 0x2,
+    //`FOP_FUSIONS` produce ops in the FusedOps category that might
+    // not be supported by all backends
+    FOP_FUSIONS = 0x4,
+    ALL_FUSIONS = 0xFFFFFFFF
+};
+using FusionTypeMask = ngraph::EnumMask<FusionType>;
```
Comment (GlebKazantaev) [inline ngraph/core/include/openvino/pass/pass.hpp:108]: let's remove this

**Pull request merged**